okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java

```diff
@@ -98,6 +92,36 @@ public void start(Handler<AsyncResult<Void>> startFuture) {
     }
   }
 
+  private static Process launch(Vertx vertx, String id, EnvEntry[] env,
+                                String [] command) throws IOException {
+
+    ProcessBuilder pb = new ProcessBuilder(command);
+    if (env != null) {
+      Map<String, String> penv = pb.environment();
+      for (EnvEntry nv : env) {
+        penv.put(nv.getName(), nv.getValue());
+      }
+    }
+    Process process = pb.start();
+    captureStream(vertx, id, process.getInputStream());
+    captureStream(vertx, id, process.getErrorStream());
+    return process;
+  }
+
+  private static void captureStream(Vertx vertx, String id, InputStream input) {
+    vertx.executeBlocking(res -> {
```

Comment:
The default worker pool size is 20. If I start 11 processes and each runs 2 captureStream there is some blocking. If the first runs 60 seconds and all other run 1 second then the logging of the 11th is blocked until the first has finished.

https://vertx.io/docs/vertx-core/java/#blocking_code says about executeBlocking:
> Blocking code should block for a reasonable amount of time (i.e no more than a few seconds). Long blocking operations or polling operations (i.e a thread that spin in a loop polling events in a blocking fashion) are precluded. When the blocking operation lasts more than the 10 seconds, a message will be printed on the console by the blocked thread checker. Long blocking operations should use a dedicated thread managed by the application, which can interact with verticles using the event-bus or runOnContext

See also https://github.com/vietj/childprocess-vertx-ext based on https://github.com/brettwooldridge/NuProcess :
> A low-overhead, non-blocking I/O, external Process execution implementation for Java. It is a replacement for java.lang.ProcessBuilder and java.lang.Process.
> Have you ever been annoyed by the fact that whenever you spawn a process in Java you have to create two or three "pumper" threads (for every process) to pull data out of the stdout and stderr pipes and pump data into stdin? If your code starts a lot of processes you can have dozens or hundreds of threads doing nothing but pumping data.

Reply:
Thx. I'll look for another capture approach.